src/tutorials.ts

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) LSST DM/SQuaRE
+// Distributed under the terms of the MIT License.
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+import { showDialog, Dialog } from '@jupyterlab/apputils';
+
+import { PageConfig } from '@jupyterlab/coreutils';
+
+import { IDocumentManager } from '@jupyterlab/docmanager';
+
+import { ServerConnection } from '@jupyterlab/services';
+
+import { Menu } from '@lumino/widgets';
+
+//import { ServiceManager, ServerConnection } from '@jupyterlab/services';
+import { IMainMenu } from '@jupyterlab/mainmenu';
+
+import * as token from './tokens';
+
+enum Actions {
+  COPY = 'copy',
+  FETCH = 'fetch'
+}
+
+enum Dispositions {
+  PROMPT = 'prompt',
+  OVERWRITE = 'overwrite',
+  ABORT = 'abort'
+}
+
+interface ITutorialsEntryResponse {
+  menu_name: string;
+  action: Actions;
+  disposition: Dispositions;
+  parent: string;
+  src: string;
+  dest: string;
+}
+
+class TutorialsEntry implements ITutorialsEntryResponse {
+  menu_name: string;
+  action: Actions;
+  disposition: Dispositions;
+  parent: string;
+  src: string;
+  dest: string;
+
+  constructor(inp: ITutorialsEntryResponse) {
+    (this.menu_name = inp.menu_name),
+      (this.action = inp.action),
+      (this.disposition = inp.disposition),
+      (this.parent = inp.parent),
+      (this.src = inp.src),
+      (this.dest = inp.dest);
+  }
+}
+
+interface ITutorialsHierarchyResponse {
+  entries: { [name: string]: ITutorialsEntryResponse } | null;
+  subhierarchies: { [name: string]: ITutorialsHierarchyResponse } | null;
+}
+
+class TutorialsHierarchy implements ITutorialsHierarchyResponse {
+  entries: { [name: string]: TutorialsEntry } | null = null;
+  subhierarchies: { [name: string]: TutorialsHierarchy } | null = null;
+
+  constructor(inp: ITutorialsHierarchyResponse, name: string | null = null) {
```

Comment:
Might be personal preference but I'd use methods outside of the constructor like:

```
  constructor(
    input: ITutorialsHierarchyResponse,
    private readonly name: string = '<unnamed>'
  ) {
    console.log(`Building hierarchy ${this.name}`);
    this.processEntries(input.entries);
    this.processSubhierarchies(input.subhierarchies);
    console.log(`Hierarchy ${this.name} built`);
  }
```